relay-server/src/services/processor.rs

```diff
@@ -1852,7 +1859,11 @@ impl EnvelopeProcessorService {
         for bucket in &mut buckets {
             clock_drift_processor.process_timestamp(&mut bucket.timestamp);
             if !keep_metadata {
-                bucket.metadata = BucketMetadata::new();
+                bucket.metadata = BucketMetadata::default();
+            }
+
+            if override_received_at_metadata {
```

Comment:
We purposefully override the metadata after the decision to keep it has been made.